modules/marshaller-common/src/main/java/org/apache/ignite/internal/marshaller/MarshallerReader.java

```diff
@@ -30,6 +30,18 @@
  * Binary reader.
  */
 public interface MarshallerReader {
+    /**
+     * Set reader's position to the given value.
+     */
+    void setIndex(int index);
```

Comment:
if we want to extend MarshallerReader with ability to read arbitrary field, it's better to extend every `read` method with field index param rather that change its internal state every time.

Reply:
Done.